### DISCUSSION TITLE
API additions for Nostr Wallet Connect (NWC) Support

### DISCUSSION THREAD

**Discussion by braydonf** [ACINQ/phoenixd#55] [Ideas]
~As part of implementing Nostr Wallet Connect for `phoenixd`, I need to return the `preimage` in the response for the `pay_invoice` command (https://github.com/nostr-protocol/nips/blob/master/47.md#pay_invoice). This however is not available in the response for `payinvoice`.~

~This could be added here:
https://github.com/ACINQ/phoenixd/blob/master/src/commonMain/kotlin/fr/acinq/lightning/bin/Api.kt#L174~

**Comment by braydonf** [comment]
There are some other values and parameters that will be needed for Nostr Wallet Connect support, for a complete list (afaik):

- ~`preimage` value to `payinvoice` response~
- ~`expiry` parameter for `createinvoice` (https://github.com/nostr-protocol/nips/blob/master/47.md#make_invoice)~
- ~`paymentHash` parameter for lookup of outgoing payments (https://github.com/nostr-protocol/nips/blob/master/47.md#lookup_invoice)~
- ~`blockheight`~ and `blockhash` values for `getinfo`, possibly also `pubkey` however I am not sure if that is for the nostr or LN public key (https://github.com/nostr-protocol/nips/blob/master/47.md#get_info)

**Reply by thesimplekid** [reply]
+1 to adding the ability to set the expiry

**Comment by braydonf** [comment]
There is also the `pay_keysend` command. Is there keysend support in Phoenix and phoenixd?

https://github.com/nostr-protocol/nips/blob/master/47.md#pay_keysend

**Comment by braydonf** [comment]
Actually, `preimage` is already available for the `payinvoice` API.

**Comment by jooray** [comment]
Is your work available somewhere? NWC support for phoenixd would be amazing.

**Comment by braydonf** [comment]
Yes it's available here: https://github.com/braydonf/satdress

**Comment by themikemoniker** [comment]
@braydonf , What is the status on adding NWC support to phoenixd? I want to help and contribute.

**Reply by Alex71btc** [reply]
you can do it with braydons sattdress or albyhub!

**Comment by braydonf** [comment]
Aside from `blockhash`, all of these are now available. Thanks!
